nixos/modules/programs/wayland/wayland-session.nix

```diff
@@ -1,23 +1,27 @@
-{ lib, pkgs, ... }: with lib; {
-    security = {
-      polkit.enable = true;
-      pam.services.swaylock = {};
-    };
+{ lib, pkgs, isHyprland ? false, xwayland ? true }:
 
-    hardware.opengl.enable = mkDefault true;
-    fonts.enableDefaultFonts = mkDefault true;
+with lib;
 
-    programs = {
-      dconf.enable = mkDefault true;
-      xwayland.enable = mkDefault true;
-    };
+{
+  security = {
+    polkit.enable = true;
+    pam.services.swaylock = {};
+  };
 
-    xdg.portal = {
-      enable = mkDefault true;
+  hardware.opengl.enable = mkDefault true;
+  fonts.enableDefaultFonts = mkDefault true;
 
-      extraPortals = [
-        # For screen sharing
-        pkgs.xdg-desktop-portal-wlr
-      ];
-    };
+  programs = {
+    dconf.enable = mkDefault true;
+    xwayland.enable = mkDefault xwayland;
+  };
+
+  xdg.portal = {
+    enable = mkDefault true;
+
+    # For screen sharing
+    extraPortals = if isHyprland
+      then [ pkgs.xdg-desktop-portal-hyprland ]
+      else [ pkgs.xdg-desktop-portal-wlr ];
```

Comment:
This should be done in the hyperland file

Reply:
Do you mean setting `xdg.portal.extraPortals` in the individual modules (Sway, River, Hyprland) instead of the common `wayland-session` one?

Reply:
I mean the condition itself should be removed as it isn't really generic